dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1332,7 +1332,14 @@ private void killAllTasks() {
             readyToSubmitTaskQueue.clear();
         }
 
+        boolean needStopProcessInstance = false;
         for (int taskId : activeTaskProcessorMaps.keySet()) {
+            if (taskRetryCheckList.containsKey(taskId)) {
+                taskRetryCheckList.remove(taskId);
+                logger.info("task id {} removed from taskRetryCheckList", taskId);
+                needStopProcessInstance = true;
```

Comment:
Why is the remove retry task associated with  needStopProcessInstance? It seems the `needStopProcessInstance` flag can be remove.